### ISSUE TITLE
Feature request. Тип ОС

### ISSUE THREAD

**Issue by asve79** [sjasmplus/sjasmplus#10]
Feature request.

В документации не нашел, создаю реквест.
Было бы здорово, если бы была системная переменная по которой можно было бы определять ОС win/lin/mac.
Это крайне полезная штука, когда разрабатываешь на разных ОС и нужно инклудить из каталогов разных форматов.

**Comment by mkoloberdin** [comment]
Не согласен что это хорошая идея. На мой взгляд незачем засорять исходный код вещами типа особенностей хост-системы кросс-ассемблера.
1. Все пути в обрабатываются с помощью библиотеки [Boost Filesystem](https://www.boost.org/doc/libs/1_69_0/libs/filesystem/doc/reference.html), т.е. можно (нужно) все пути указывать с прямыми слэшами и они будут обработаны как надо на любой системе.
2. Если надо указать глобальный путь с буквой диска на Windows, лучше всего использовать ключ командной строки `-I`, см документацию: https://github.com/sjasmplus/sjasmplus/wiki

**Comment by asve79** [comment]
По п.1 К сожалению компиляция падала. При ошибке выводила путь со смешанными слешами. Отсюда и появилась потребность в понимании типа ОС на которой идет компиляция.
По п.2 Спасибо за наводку! Буду пользоваться.

**Comment by asve79** [comment]
Подумал и  решил дописать размышления. ) Все-таки решать проблему через опцию -i не всегда получится, т к могут возникнуть ситуации, когда в разных каталога могут лежать файлы с одинаковыми названиями.

**Comment by mkoloberdin** [comment]
Не понятно как определение ОС поможет решить такую проблему. Тут речь идет о специфике расположения файлов на конкретном хосте, а не ОС.
Не лучше ли нормально разложить файлы получше, использовать относительные пути (относительно `-Ι`) + `-Ι` можно использовать несколько.
Например:
`Library1/Dir2/File3.asm`
`Library4/Dir5/File6.asm`
...
`sjasmplus -IFullPathToLibrary1 -IFullPathToLibrary4`
В исходниках:
`include "Dir2/File3.asm"`
`include "Dir5/File6.asm"`

**Comment by asve79** [comment]
Использование относительных путей на данный момент было ну совсем не очевидно. 
Это я вот взял и спросил, а кто-то почешет репу и пройдет мимо (А мог бы делать полезный софт для спека).
Нужно посмотреть как оно работает и как будет показывать файлы при ошибке (как будет выглядить путь к файлу, чтобы его смог перехватить редактор).
А переразложить файлы не всегда может быть возможным. Например если пользоваться чьими-то библиотеками в которых расположение может быть «прибито». 
Да мало ли еще зачем понадобится. ИМХО ж не так сложно добавить еще одну системную переменную которая может жестко вшиваться при сборке асма в бинарик.
